File: lib/models/appModels/Image.dart
import 'User.dart';
import 'Event.dart';
import 'Chat.dart';
import 'Team.dart';
import 'Location.dart';
import 'Organization.dart';
import 'Sponsor.dart';


class Image {
  bool? isMainImage;
  bool? public;
  String? url;
  String? key;
  User? user;
  Event? event;
  Chat? chat;
  Team? team;
  Location? location;
  Organization? organization;
  ModelType? attachedModelType;
  Sponsor? sponsor;
  String? s3bucket;

  Image({
    this.isMainImage,
    this.public,
    this.url,
    this.key,
    this.user,
    this.event,
    this.chat,
    this.team,
    this.location,
    this.organization,
    this.attachedModelType,
    this.sponsor,
    this.s3bucket,
  });
}

File: apollo/src/Image.ts
import mongoose from 'mongoose';
const Schema = mongoose.Schema;


const imageSchema = new mongoose.Schema({
    isMainImage: Boolean,
    public: Boolean,
    url: String,
    key: String,
    user: { type: Schema.Types.ObjectId, ref: 'User' },
    event: { type: Schema.Types.ObjectId, ref: 'Event' },            
    location: { type: Schema.Types.ObjectId, ref: 'Location'},          
    s3bucket: String
});

const Image = mongoose.model('Image', imageSchema);

export default Image;